### DISCUSSION TITLE
Moving the document in visual mark mode rather than the mark?

### DISCUSSION THREAD

**Discussion by panangam** [ahrm/sioyek#1226] [Q&A] [answered]
In the visual mark mode, instead of moving the highlighted line down the document, I want to keep the highlighted line in the middle of the page, and scroll the entire document accordingly. This will keep the box/underline at the center of the screen at all time. Is there a way to achieve this using existing commands? For example, is there a command that scroll the document to put the mark at a certain offset from the top?

**Comment by ahrm** [comment, marked as answer]
We do have an `autocenter_visual_scroll` config. Does that help your case?

**Reply by panangam** [reply]
Sure does 👍.

I figured out a neat hack though, so I just wanna mention here. By setting `visual_mark_next_page_threshold` to a high number, I can force the page to scroll every time, and I can set the visual mark vertical position using `visual_mark_next_page_fraction`. So right now I have 
```
visual_mark_next_page_fraction 0.5
visual_mark_next_page_threshold 2
```
which puts the mark 1/4 of the way from the top. Kinda neat.
